runtime/interpreter/interpreter.go

```diff
@@ -876,93 +880,82 @@ func (interpreter *Interpreter) VisitFunctionBlock(_ *ast.FunctionBlock) ast.Rep
 func (interpreter *Interpreter) visitFunctionBody(
 	beforeStatements []ast.Statement,
 	preConditions ast.Conditions,
-	body func() interface{},
+	body func() controlReturn,
 	postConditions ast.Conditions,
 	returnType sema.Type,
-) Trampoline {
+) controlReturn {
```

Comment:
```suggestion
) functionReturn {
```

Reply:
Right, nice! I realized that this can be further simplified to just `Value`, refactored it in d9e01a2975fdcfdfab0df8c514cf438fe5e596fe